modules/encrypt-node/src/framed_encrypt_stream.ts

```diff
@@ -190,11 +190,18 @@ type EncryptFrameInput = {
 export function getEncryptFrame (input: EncryptFrameInput): EncryptFrame {
   const { pendingFrame, messageHeader, getCipher, isFinalFrame } = input
   const { sequenceNumber, contentLength, content } = pendingFrame
-  const frameIv = serialize.frameIv(messageHeader.headerIvLength, sequenceNumber)
+  const { frameLength, contentType, messageId, headerIvLength } = messageHeader
+  /* Precondition: The content length MUST correlate with the frameLength.
+   * In the case of a regular frame,
+   * the content length must strictly equal the frame length.
+   * In the case of the final frame,
+   * it MUST not be larger than the frame length.
+   */
+  needs(frameLength === contentLength || (isFinalFrame && frameLength >= contentLength), 'Final frame length exceeds frame length.')
```

Comment:
Strictly speaking, this should be two separate checks, no? If something goes terribly wrong on a non-final frame and I get an error that's talking about final frame length, I'm gonna be confused.

Reply:
Perhaps.  What about the error message `Content length exceeds frame length`

Reply:
How about:

```suggestion
  needs(isFinalFrame || frameLength === contentLength, 'Content length does not match frame length.')
  needs(!isFinalFrame || frameLength >= contentLength, 'Final frame length exceeds frame length.')
```

Reply:
Alternate suggestion:
```suggestion
  if (isFinalFrame){
    needs(frameLength >= contentLength, 'Final frame length exceeds frame length.')
  }else{
    needs(frameLength === contentLength, 'Content length does not match frame length.')
  }
```

Reply:
Talking about this, having the exact values in question in the error message should resolve this.

Both the error message AND the condition should clearly express the intent.
Further, if this error is ever hit, knowing if you are a final frame or not will GREATLY simplify the debugging.